apps/userdashboard/assets/js/a4_candy_userdashboard/FilterBar.jsx

```diff
@@ -1,29 +1,49 @@
-import React from 'react'
+import React, { useState } from 'react'
 import django from 'django'
 
-export const FilterBar = () => {
+const getItemByFilter = (items, filter) => {
+  return items.find(item => item.filter === filter)
+}
+
+export const FilterBar = (props) => {
+  const filterItems = [
+    { name: 'Pending', filter: '?is_pending=true' },
+    { name: 'Archived', filter: '?is_pending=false' },
+    { name: 'All', filter: '' }
+  ]
+  const [currFilterItem, setCurrFilterItem] =
+    useState(getItemByFilter(filterItems, props.selectedFilter))
+
+  const onSelectFilter = (filterItem) => {
+    setCurrFilterItem(filterItem)
+    props.onFilterChange(filterItem.filter)
+  }
+
   return (
-    <div className="filter-bar justify-content-end" role="group" aria-label={django.gettext('Filter')}>
-      <div className="me-sm-1 mt-2 mt-sm-0">
+    <div
+      className="filter-bar justify-content-end"
+      role="group"
+      aria-label={django.gettext('Filter')}
+    >
+      <div className="mt-2 mt-sm-0">
         <div className="dropdown dropdown-menu-end">
           <button
             type="button"
-            className="dropdown-toggle btn btn--light btn--select show"
+            className="dropdown-toggle btn btn--light btn--select"
             data-bs-toggle="dropdown"
+            aria-expanded="false"
           >
-            {django.gettext('Filter')}
+            {django.gettext('Filter')}: {currFilterItem.name}
             <i className="fa fa-caret-down" aria-hidden="true" />
           </button>
           <ul className="dropdown-menu">
-            <li>
-              <a className="selected" href="/#">all</a>
-            </li>
-            <li>
-              <a href="/#">pending</a>
-            </li>
-            <li>
-              <a href="/#">archived</a>
-            </li>
+            {filterItems.map((filterItem, i) => (
+              <li key={`${i}_${filterItem.name}`}>
```

Comment:
this might be a super dumb question but why do you use `$` here?

Reply:
ah, i am using `template literals` here. meaning whatever is in between `${ .. }` will be treated as expression.
A better example would be:

`Hello ${name}, welcome to this.`